bootstrap-gruntwork-installer.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+#
+# A bootstrap script to install the Gruntwork Installer.
+#
+# Why:
+#
+# The goal of the Gruntwork Installer is to make make installing Gruntwork Script Modules feel as easy as installing a
+# package using apt-get, brew, or yum. However, something has to install the Gruntwork Installer first. One option is
+# for each Gruntwork client to do so manually, which would basically entail copying and pasting all the code below.
+# This is tedious and would give us no good way to push updates to this bootstrap script.
+#
+# So instead, we recommend that clients use this tiny bootstrap script as a one-liner:
+#
+# curl -Ls https://raw.githubusercontent.com/gruntwork-io/gruntwork-installer/master/bootstrap-gruntwork-installer.sh | bash -s --version=0.0.12
+#
+# You can copy this one-liner into your Packer and Docker templates and immediately after, start using the
+# gruntwork-install command.
+
+set -e
+
+readonly DEFAULT_FETCH_VERSION="v0.0.3"
+readonly FETCH_DOWNLOAD_URL_BASE="https://github.com/gruntwork-io/fetch/releases/download"
+readonly FETCH_INSTALL_PATH="/usr/local/bin/fetch"
+
+readonly GRUNTWORK_INSTALLER_DOWNLOAD_URL_BASE="https://raw.githubusercontent.com/gruntwork-io/gruntwork-installer"
+readonly GRUNTWORK_INSTALLER_INSTALL_PATH="/usr/local/bin/gruntwork-install"
+readonly GRUNTWORK_INSTALLER_SCRIPT_NAME="gruntwork-install"
+
+function print_usage {
+  echo
+  echo "Usage: bootstrap-gruntwork-installer.sh [OPTIONS]"
+  echo
+  echo "A bootstrap script to install the Gruntwork Installer ($GRUNTWORK_INSTALLER_SCRIPT_NAME)."
+  echo
+  echo "Options:"
+  echo
+  echo -e "  --version\t\tRequired. The version of $GRUNTWORK_INSTALLER_SCRIPT_NAME to install (e.g. 0.0.3)."
+  echo -e "  --fetch-version\tOptional. The version of fetch to install. Default: $DEFAULT_FETCH_VERSION."
+  echo
+  echo "Examples:"
+  echo
+  echo "  Install version 0.0.3:"
+  echo "    bootstrap-gruntwork-installer.sh --version=0.0.3"
+  echo
+  echo "  One-liner to download this bootstrap script from GitHub and run it to install version 0.0.3:"
+  echo "    curl -Ls https://raw.githubusercontent.com/gruntwork-io/gruntwork-installer/master/bootstrap-gruntwork-installer.sh | bash -s --version=0.0.3"
+}
+
+function command_exists {
+  local readonly cmd="$1"
+  type "$cmd" > /dev/null 2>&1
+}
+
+function download_url_to_file {
+  local readonly url="$1"
+  local readonly file="$2"
+
+  echo "Downloading $url to $file"
+  if $(command_exists "curl"); then
+    local readonly status_code=$(curl -L -s -w '%{http_code}' -o "$file" "$url")
+    if [[ "$status_code" != "200" ]]; then
+      echo "ERROR: Expected status code 200 but got $status_code when downloading $url"
+      exit 1
+    fi
```

Comment:
I had to add this as I messed up the URL, but got no error, as the web page returned a 404, and curl sees that as a successful download. I also recently updated the code to download to a tmp file first, and only copy to the final destination if it got a 200 OK.

We should consider using this function everywhere we use `curl` in bash scripts.